src/asn1sccdocumentprocessor.cpp

```diff
@@ -33,20 +33,21 @@ Asn1SccDocumentProcessor *Asn1SccDocumentProcessor::create(const QString &projec
 {
     return new Asn1SccDocumentProcessor(projectName,
                                         [](const QHash<QString, QString> &documents)->ParsedDocumentBuilder *
-                                        { return Asn1SccParsedDocumentBuilder::create(documents); });
+                                        { return Asn1SccParsedDocumentBuilder::create(documents); },
+                                        ParsedDataStorage::instance());
 }
 
 Asn1SccDocumentProcessor::Asn1SccDocumentProcessor(const QString &projectName,
-                                                   DocumentBuilderCreator docBuilderCreator)
-    : m_projectName(projectName),
-      m_state(State::Unfinished),
-      m_docBuilder(nullptr),
-      m_docBuilderCreator(docBuilderCreator)
-{
-}
-
-Asn1SccDocumentProcessor::~Asn1SccDocumentProcessor()
+                                                   DocumentBuilderCreator docBuilderCreator,
+                                                   ParsedDataStorage *storage)
+    : m_projectName(projectName)
+    , m_state(State::Unfinished)
+    , m_docBuilder(nullptr)
+    , m_docBuilderCreator(docBuilderCreator)
+    , m_storage(storage)
 {
+    m_index = m_storage->getProjectBuildersCount(m_projectName) + 1;
```

Comment:
does separate index for each project is really needed? I do not want that to change, but it's not the simplest solution and I wonder why :)

Reply:
Hmm, single project build is matched with one request to asn1scc daemon. I think requests should be concurrent within project, otherwise (in case one global index for all requests) the idea with numbering would not work when file modified would belong to more than one project.

Reply:
ack